### ISSUE TITLE
Error creating bean with name 'kubernetesClient': Requested bean is currently in creation

### ISSUE THREAD

**Issue by jhaeyaert** [gravitee-io/issues#8927]
## Describe the bug

When setting `services.metrics.enabled` to `true`, and specifying a Kubernetes value in the configuration, the node (management or gateway, apim or am) fails with the following error:

<div class="preformatted panel" style="border-width: 1px;"> 
 <div class="preformattedContent panelContent"> 
  <pre>at io.gravitee.node.vertx.VertxFactory.getObject(VertxFactory.java:53)
at org.springframework.beans.factory.support.FactoryBeanRegistrySupport.doGetObjectFromFactoryBean(FactoryBeanRegistrySupport.java:169)
... 87 more
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'io.gravitee.node.kubernetes.propertyresolver.KubernetesPropertyResolver': Unsatisfied dependency expressed through field 'kubernetesClient'; nested exception is org.springframework.beans.factory.BeanCurrentlyInCreationException: Error creating bean with name 'kubernetesClient': Requested bean is currently in creation: Is there an unresolvable circular reference?
at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.resolveFieldValue(AutowiredAnnotationBeanPostProcessor.java:660)
at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:640)
at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:119)
at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessProperties(AutowiredAnnotationBeanPostProcessor.java:399)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1431)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireBean(AbstractAutowireCapableBeanFactory.java:343)
at io.gravitee.common.spring.factory.SpringFactoriesLoader.createSpringFactoriesInstances(SpringFactoriesLoader.java:90)
... 113 more
Caused by: org.springframework.beans.factory.BeanCurrentlyInCreationException: Error creating bean with name 'kubernetesClient': Requested bean is currently in creation: Is there an unresolvable circular reference?
at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.beforeSingletonCreation(DefaultSingletonBeanRegistry.java:355)
at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:227)
at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)
at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276)
at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1391)
at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1311)
at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.resolveFieldValue(AutowiredAnnotationBeanPostProcessor.java:657)
... 119 more</pre> 
 </div> 
</div>

## To Reproduce

Steps to reproduce the behaviour:

1.  Configure the environment variable gravitee.services.metrics.enabled=true
2.  Add a property with a value starting with kuberntes:// (ex: `gravitee.ratelimit.type=kubernetes://ratelimit/type`)
3.  Start the apim gateway and see the error.

## Expected behaviour

The server should start without error.

## Current behaviour

The server fails to start with the above error.

## Useful information

N/A

## Desktop:

**_Please complete the following information:_** 

*   Environment: APIM 3.19 and above, AM 3.21
*   OS: N/A
*   BrowserN/A

## Potential impacts

**_Which other features may be impacted by this fix. This could be populated after fix_** 

**_What are the impacted versions?_** 

APIM 3.19 and above, AM 3.21 (only?).

Basically, all versions that include gravitee-node > v1.25.0
